src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java

```diff
@@ -77,6 +91,37 @@ public void rebuildSessionFromFrame(MenuSession menuSession) throws CommCareSess
                         break;
                     }
                 }
+            } else if (screen instanceof QueryScreen) {
+                QueryScreen queryScreen = (QueryScreen)screen;
+                RemoteQueryDatum neededDatum = (RemoteQueryDatum) menuSession.getSessionWrapper().getNeededDatum();
+                boolean done = false;
+                for (StackFrameStep step : steps) {
+                    if (step.getId().equals(neededDatum.getDataId())) {
+                        URI uri = null;
+                        try {
+                            uri = new URI(step.getValue());
+                        } catch (URISyntaxException e) {
```

Comment:
Echoing @shubham1g5's feedback in commcare-core that it would probably be good to push this validation to parse time (we can still and wrap here, but will push any **expected** failures to a point where they'd be caught in a cleaner state.

Reply:
https://github.com/dimagi/commcare-core/pull/1048/commits/d1ea42af5e893cb33a097583fd40c21b89027ade